pkg/controller/installation/typha_autoscaler.go

```diff
@@ -157,15 +198,18 @@ func (t *typhaAutoscaler) autoscaleReplicas() error {
 	if err != nil {
 		return fmt.Errorf("could not get number of nodes: %w", err)
 	}
-	expectedReplicas := utils.GetExpectedTyphaScale(allSchedulableNodes)
+	log.V(1).Info("Number of nodes to consider for typha autoscaling", "all", allSchedulableNodes, "linux", linuxNodes)
+	expectedReplicas := common.GetExpectedTyphaScale(allSchedulableNodes)
 	if linuxNodes < expectedReplicas {
 		return fmt.Errorf("not enough linux nodes to schedule typha pods on, require %d and have %d", expectedReplicas, linuxNodes)
 	}
 
-	err = t.updateReplicas(int32(expectedReplicas))
-
-	if err != nil && !apierrors.IsNotFound(err) {
-		return fmt.Errorf("could not scale Typha deployment: %w", err)
+	log.V(1).Info("Checking if we need to scale typha", "expectedReplicas", expectedReplicas, "currentReplicas", t.activeReplicas)
+	if int32(expectedReplicas) != t.activeReplicas {
```

Comment:
This check allows us to skip the Get() call within updateReplicas if we don't actually need to update the replicas. We'll spot when the number of replicas changes because we're instead watching the typha deployment.